[master][CC] Adding docs for deploying config changes into Choreo Connect (#6126)
wso2/docs-apim: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md

#### Step 5: Deploy Choreo Connect

> Please follow the document about [Deploying Choreo Connect as a Standalone Gateway on Kubernetes - Helm Artifacts]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts/) for deploying Choreo Connect.

Reviewer: ```suggestion
Follow the document on [Deploying Choreo Connect as a Standalone Gateway on Kubernetes - Helm Artifacts]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts/) for deploying Choreo Connect.
```

> ### Applying Config Changes into Running Instance of Choreo Connect 

Reviewer: ```suggestion
### Applying config changes into a running instance of Choreo Connect 
```

When you have to deploy a config change to the Choreo Connect running on production environment, we recommand you to complete the following steps in order.

!!! Attention

>     You must follow this, if the change of the config is related to **enforcer** as `enforcer` is fetching configs from adapter only at the startup.

Reviewer: ```suggestion
    You must follow this, if the config change is related to the **Enforcer** as the `enforcer` fetches configs from the Adapter only at the startup.
```

Steps:

> 1. Do the config change and first rollout restart the adapter.

Reviewer: ```suggestion
1. Do the config change and first rollout restart the Adapter.
```

> 2. After all replicas of adapter pod are started and ready, rollout restart enforcer & router (`choreo-connect-deployment`).

Reviewer: ```suggestion
2. After all the replicas of the Adapter pod have been started and are ready, rollout restart the Enforcer and Router (`choreo-connect-deployment`).
```

> 2. After all the replicas of the Adapter pod have been started and are ready, rollout restart the Enforcer and Router (`choreo-connect-deployment`).

Reviewer: What do we mean by "rollout restart"? Are there some different steps to do this compared to normal restart?

Author: So this production development guide is based on deploying things in kubernetes.. what I meant by "rolling restart" is the default deployment strategy of kubernetes.

Reviewer: Shall we add that context here?

Reviewer 2: Maybe we can link to https://kubernetes.io/docs/reference/kubectl/cheatsheet/#updating-resources

Author: Linked to the above..
